src/main/java/com/codewars/junit5/CodewarsListener.java

```diff
@@ -21,193 +21,214 @@
+                        outputFailure("Failed", th.get());
+                    } else {
+                        System.out.println("\n<FAILED::>Failed for unknown cause");
+                    }
+                } else {
+                    ++failures;
+                    Optional<Throwable> th = testExecutionResult.getThrowable();
+                    if (th.isPresent()) {
+                        outputError("Crashed", th.get());
+                    } else {
+                        System.out.println("\n<ERROR::>Unexpected error occurred");
+                    }
+                }
+                System.out.printf("\n<COMPLETEDIN::>%d\n", getDuration(testIdentifier));
+                break;
+
+            default:
+                throw new Error("Unsupported execution status:" + testExecutionResult.getStatus());
+        }
+    }
+
+    // > In JUnit Jupiter you should use `TestReporter` where you used to print
+    // information to `stdout` or `stderr` in JUnit 4.
+    // ```java
+    // @Test
+    // void reportSingleValue(TestReporter testReporter) {
+    // testReporter.publishEntry("a key", "a value");
+    // }
+    // ```
+    public void reportingEntryPublished(TestIdentifier testIdentifier, ReportEntry entry) {
+        reportEntries(testIdentifier).add(reportEntryToString(entry));
+    }
+
+    public int failures() {
+        return failures;
+    }
+
+    public int testCount() {
+        return testCount;
+    }
+
+    private static void outputFailure(String kind, Throwable throwable) {
+        String msg = throwable.getMessage();
+        if (msg == null) {
+            System.out.printf("\n<FAILED::>Test %s\n", kind);
+        } else {
+            System.out.printf("\n<FAILED::>%s\n", formatMessage(msg));
         }
-        System.out.printf("\n<COMPLETEDIN::>%d\n", getDuration(testIdentifier));
-        break;
-
-      case ABORTED: // assumptions not met
-        if (testIdentifier.isTest()) {
-          ++failures;
-          Optional<Throwable> th = testExecutionResult.getThrowable();
-          if (th.isPresent()) {
-            outputFailure("Aborted", th.get());
-          } else {
-            System.out.println("\n<FAILED::>Aborted for unknown cause");
-          }
+        System.out.printf("\n<LOG:ESC:-Stack Trace>%s\n", formatMessage(readStackTrace(throwable)));
+    }
+
+    private static void outputError(String kind, Throwable throwable) {
+        String msg = throwable.getMessage();
+        if (msg == null) {
+            System.out.printf("\n<ERROR::>Test %s\n", kind);
+            System.out.printf("\n<LOG:ESC:-Stack Trace>%s\n", formatMessage(readStackTrace(throwable)));
+        } else {
+            String formattedMessage = formatMessage(msg);
+            String formattedStackTrace = formatMessage(readStackTrace(throwable));
+            System.out.printf("\n<ERROR::>%s<:LF:><:LF:>Stack trace:<:LF:><:LF:>%s\n", formattedMessage, formattedStackTrace);
```

Comment:
Is there a reason not to collapse the stack trace like above (L176)?

Reply:
Well my reasoning was that a collapsed stacktrace does not necessarily represent a (potentially severe) problem with runtime but more a failing test case. I thought that since it's an internal error of JUnit, it deserves to crash hard. But it's just what I thought, can change it to whatever you prefer.

Reply:
Makes sense. Can you still put them in a separate container? Just don't add `-`. I think that's more readable.

Reply:
Here it is. I am not sure I like how the stacktrace now looks logically separate from the error, but if you prefer it this way, then fine.  
BTW stacktraces in log panel look much better than in the `ERROR::` panel due to lack of excessive line wrapping. Would it be possible to make the error panel behave in similar way as log panels?

Reply:
Can you post a screenshot? I saw the original on Discord, but can you also repost that here, so it's easier to compare?

I'll look into the excessive line wrapping.

Reply:
Not sure what screenshot exactly you want? Here's the `ERROR` with `LOG` vs a single `ERROR`:

![image](https://user-images.githubusercontent.com/23709795/216200285-e953d02a-7c9f-42c6-9c00-b5ef76430ca9.png)

![image](https://user-images.githubusercontent.com/23709795/216200454-7ee20bcc-7a84-4502-b48c-65b72ed7edde.png)


Is this what you asked?

Reply:
For comparison, here is with CSS property `white-space: pre`:

![image](https://user-images.githubusercontent.com/23709795/216201150-7031dbd6-8b3b-4115-aca1-1c1e0799d4a5.png)

Reply:
Yeah, the last one might be the best. I think "Stack Trace:" can be removed.

I'll fix the CSS and let you know.

Reply:
Deployed the CSS change.